docs/index.rst

```diff
@@ -88,3 +71,20 @@ Open3D: A Modern Library for 3D Data Processing
     python_example/pipelines/index
     python_example/utility/index
     python_example/visualization/index
+
```

Comment:
As previous commit, this one focuses on putting all python related stuff together, i.e. `Tutorials`, `Python API` and `Python Examples`. C++ API after Python and Contribution in the end. 

![Screenshot from 2023-08-12 18-01-55](https://github.com/isl-org/Open3D/assets/11634769/fbb6d5fb-dfbf-49e8-8a64-e1f704d36835)

![Screenshot from 2023-08-12 18-02-09](https://github.com/isl-org/Open3D/assets/11634769/c3a6e1d9-1fe8-4f1a-b281-70cb61f37d7b)